lib: nrf_cloud: Fix P-GPS prediction handling (#28497)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * Removed dependency on the :ref:`lte_lc_readme` library.
    The :ref:`at_monitor_readme` library is now used for tracking network connectivity status.

* :ref:`lib_nrf_cloud_pgps` library:

  * Updated the range for the :kconfig:option:`CONFIG_NRF_CLOUD_PGPS_NUM_PREDICTIONS` and :kconfig:option:`CONFIG_NRF_CLOUD_PGPS_REPLACEMENT_THRESHOLD` Kconfig options to values supported by nRF Cloud.

  * Fixed:

>     * An issue which caused predictions to be stored into incorrect index in flash, leading to an error when the affected predictions were tried to be used.

Reviewer: ```suggestion
    * An issue that caused predictions to be stored into incorrect index in flash, leading to an error when the affected predictions were used.
```

>     * An issue where the library tried to use predictions which had not yet been flushed to flash, leading to a prediction validation error.

Reviewer: ```suggestion
    * An issue where the library tried to use predictions that had not yet been flushed to flash, leading to a prediction validation error.
```
